packages/expo-router/src/__tests__/navigation.test.ios.tsx

```diff
@@ -810,6 +810,68 @@ it('can push the same route multiple times', () => {
   expect(screen.getByTestId('index')).toBeOnTheScreen();
 });
 
+it('can continue to push with same path but different params', async () => }
+  renderRouter({
```

Comment:
This test is pass on the current `main` branch without your changes. I'm not sure it correctly describes the behaviour your fixing.

Reply:
I think I had to change it to a drawer. Commit: https://github.com/expo/expo/pull/26597/commits/7f79de417f650675bd2752ae44d92a2eb0671857